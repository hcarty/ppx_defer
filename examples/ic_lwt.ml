let%lwt () =
  let%lwt ic = Lwt_io.open_file ~mode:Lwt_io.input "README.md" in
  [%defer.lwt Lwt_io.close ic];
  let%lwt bytes = Lwt_io.read ic in
  Lwt_io.printl bytes
