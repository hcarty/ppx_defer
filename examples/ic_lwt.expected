let promise =
  let%lwt ic = Lwt_io.open_file ~mode:Lwt_io.input "ic_lwt.ml" in
  [%defer.lwt Lwt_io.close ic];
  let%lwt bytes = Lwt_io.read ic in
  Lwt_io.printl bytes

let () = Lwt_main.run promise

